db/migrate/20240126213348_set_existing_form_state.rb

```diff
@@ -0,0 +1,23 @@
+class SetExistingFormState < ActiveRecord::Migration[7.1]
+  def up
+    Form.find_each do |form|
+      state = if form.has_live_version
+                if form.has_draft_version
+                  :live_with_draft
+                else
+                  :live
+                end
+              else
+                :draft
+              end
+
+      form.update_column(:state, state)
```

Comment:
tested this locally by creating three forms in draft, live & live_with_draft state using the main branch and then switching to this branch to run the migration. The 3 forms states were correctly set and none of the `updated_at` columns changed